tasks/configure.yml

```diff
@@ -78,8 +83,26 @@
 - name: Flush handlers to ensure Zookeeper is up to date
   meta: flush_handlers
 
-- name: Wait for Zookeeper port
+- name: Wait for Zookeeper client port
   wait_for:
     port: "{{ zookeeper.client_port }}"
     state: started
-    timeout: 30
+    timeout: 60
+
+- name: Copy zookeeper check script
+  become: yes
+  become_user: "{{ zookeeper.user }}"
+  copy:
+    src: zookeeper_check
+    dest: "/home/{{ zookeeper.user }}/bin/zookeeper_check"
+    owner: "{{ zookeeper.user }}"
+    group: "{{ zookeeper.group }}"
+    mode: 0750
+
+- name: Run zookeeper check script
+  become: yes
+  become_user: "{{ zookeeper.user }}"
+  shell: "/home/{{ zookeeper.user }}/bin/zookeeper_check"
+  async: 7200
+  poll: 0
```

Comment:
It runs in background until is successful. Sometimes takes 2 loops sometimes 3. This will extend even more Pause time (signal from instance). We use 'Shared tenancy' so performance is not always great.